### DISCUSSION TITLE
How can I collect metrics for monitoring xrdp activity?

### DISCUSSION THREAD

**Discussion by NinhDoan** [neutrinolabs/xrdp#3147] [Q&A]
Hi, I am currently looking to collect metrics for monitoring the activity of xrdp. Could you please provide guidance or examples on how to achieve this?

**Comment by matt335672** [comment]
For network/CPU/memory/authentication I'd suggest using the standard OS tools.

We don't provide much that I'm aware of as part of the product itself other than logging support for `fail2ban`.

Is there anything specific you want to look at?

**Reply by NinhDoan** [reply]
I'm looking to collect metrics for RDP in xrdp, such as frames per second and round-trip latency for frame packets. It would also be very helpful to have metrics for frame loss or skips. I'm deploying a series of servers running xrdp, so I need to monitor their activity. I couldn't find any guidance on this, so any help would be appreciated. Thank you.

**Reply by matt335672** [reply]
That's not something we provide directly.

It may not be quite what you want anyway. A 'frame' in an RDP sense is just a set of updates that have been collected together - it doesn't correspond to a standard monitor definition of frames per second. 

There's scope for measuring the RTT as part of the RDP spec (see #2396) but it's not something we currently implement.
